internal/restorer/restorer.go

```diff
@@ -511,12 +511,32 @@ func (res *Restorer) removeUnexpectedFiles(ctx context.Context, target, location
 		selectedForRestore, _ := res.SelectFilter(nodeLocation, false)
 		// only delete files that were selected for restore
 		if selectedForRestore {
-			res.opts.Progress.ReportDeletedFile(nodeLocation)
+			// First collect all files that will be deleted
+			var filesToDelete []string
+			err := filepath.Walk(nodeTarget, func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+				if !info.IsDir() {
```

Comment:
`ReportDeletedFile` previously also reported deletions of directories. I think we should keep that behavior. That is, just collect all directory entries that would be deleted. And ideally rename `ReportDeletedFile` to `ReportDeletion`.

Reply:
Done. 
1. I have removed the directory check. 
2. I have renamed the function to `ReportDeletion`